Template.php

```diff
@@ -1388,15 +1388,10 @@ protected function google_book_details ($gid) {
     if ($this->blank("editor") && $this->blank("editor1") && $this->blank("editor1-last") && $this->blank("editor-last") && $this->blank("author") && $this->blank("author1") && $this->blank("last") && $this->blank("last1") && $this->blank("publisher")) { // Too many errors in gBook database to add to existing data.   Only add if blank.
       foreach ($xml->dc___creator as $author) {
         if( in_array(strtolower($author), BAD_AUTHORS) === FALSE) {
-          if( in_array(strtolower($author), AUTHORS_ARE_PUBLISHERS) === TRUE  ||
-             substr(strtolower($author),-4)  === " inc" || 
-             substr(strtolower($author),-5)  === " inc." ||
-             substr(strtolower($author),-10) === " magazines" || 
-             substr(strtolower($author),-6)  === " press" ||
-             substr(strtolower($author),-6)  === " books" ||
-             substr(strtolower($author),-11) === " publishing" ||
-             substr(strtolower($author),-11) === " publishers" ||
-             substr(strtolower($author),-12) === " corporation") {
+          $author_parts  = explode(" ",$author);
+          $author_ending = ' ' . end($author_parts);
+          if( in_array(strtolower($author),       AUTHORS_ARE_PUBLISHERS        ) === TRUE ||
+              in_array(strtolower($author_ending),AUTHORS_ARE_PUBLISHERS_ENDINGS) === TRUE) {
```

Comment:
Another point for the style guide: commas should always be followed by a space, if that's okay.

Reply:
Done!